src/imp.rs

```diff
@@ -526,6 +509,45 @@ where
     }
 }
 
+/// Pick the notes to be used in a transaction from a vector of notes.
+///
+/// The notes are picked in a way to maximize the number of notes used, while
+/// best-effort minimizing the value employed. To do this we sort the notes in
+/// ascending value order, and go through each contiguous slice of the largest
+/// possible size until we find the first slice whose sum is larger or equal to
+/// the given value. If such a slice is not found, None is returned.
+///
+/// Note: it is presupposed that the input notes contain enough balance to cover
+/// the given `value`.
+fn pick_notes(
+    value: u64,
+    mut notes_and_values: Vec<(Note, u64, JubJubScalar)>,
+) -> Option<Vec<(Note, u64, JubJubScalar)>> {
```

Comment:
I see a couple of API issue here. The most important one is, we're taking a mutable ownership of the `notes_and_values` and we returns the same type; and we have an `Option<Vec<>>`.

And about the mutable ownership, we might want to have a mutable reference and then "reduce" the inputs to the outputs we want – and we remove the return value – or we could have a reference to the input (so, no mutation) and we could return a new value.
Having this signature feels a bit akward IMVHO.

I believe that in this case, an empty vector would be equivalent to `None`, so we might remove the `Option` here and we can just check `is_empty()`.